apps/kahoot-it-clone/src/app/signup/signup.component.ts

```diff
@@ -0,0 +1,31 @@
+import { Component } from '@angular/core';
+import {
+  FormBuilder,
+  FormGroup,
+  FormControl,
+  Validators
+} from '@angular/forms';
+import { User } from '@kahoot-it-clone/shared-types';
+import { AuthService } from '../auth-service/auth.service';
+
+@Component({
+  selector: 'kahoot-it-clone-signup',
+  templateUrl: './signup.component.html',
+  styleUrls: ['./signup.component.css']
+})
+export class SignupComponent {
+  loginForm: FormGroup;
+
+  constructor(private fb: FormBuilder, private authService: AuthService) {
+    this.loginForm = this.fb.group({
+      username: new FormControl('', Validators.required),
+      email: new FormControl('', Validators.required),
+      password: new FormControl('', Validators.required)
+    });
```

Comment:
https://angular.io/api/forms/Validators There are validators for email and min length for the password. You can even use the pattern validator to make the password include a number or special character if you want, but that one isn't necessary.